src/psycop_feature_generation/loaders/raw/utils.py

```diff
@@ -280,3 +280,62 @@ def load_from_codes(
             source_timestamp_col_name: "timestamp",
         },
     )
+
+
+def unpack_intervals(
+    df: pd.DataFrame,
+    starttime_col: str = "datotid_start_sei",
+    endtime_col: str = "timestamp",
+    unpack_freq: str = "D",
+) -> pd.DataFrame:
+    """Transform df with starttime_col and endtime_col to day grain (one row per day in the interval starttime_col-endtime_col).
+    First and last day will have the specific start and end time, while days inbetween will be 00:00:00.
+
+    Args:
+        df (pd.DataFrame): dataframe with time interval in separate columns.
+        starttime_col (str, optional): Name of column with start time. Defaults to "datotid_start_sei".
+        endtime_col (str, optional): Name of column with end time. Defaults to "timestamp".
+        unpack_freq: Frequency string by which the interval will be unpacked. Default to "D" (day). For e.g., 5 hours, write "5H".
+
+    Returns:
+        pd.DataFrame: Dataframe with time interval unpacked to day grain.
+
+    """
+
+    # create rows with end time
+    df_end_rows = df.copy()
+    df_end_rows["date_range"] = df_end_rows[f"{endtime_col}"]
+
+    # create a date range column between start date and end date for each visit/admission/coercion instance
+    df["date_range"] = df.apply(
+        lambda x: pd.date_range(
+            start=x[f"{starttime_col}"],
+            end=x[f"{endtime_col}"],
+            freq=unpack_freq,
+        ),
+        axis=1,
+    )
+
+    # explode the date range column to create a new row for each date in the range
+    df = df.explode("date_range")
+
+    # concat df with start and end time rows
+    df = pd.concat([df, df_end_rows], ignore_index=True).sort_values(
+        ["dw_ek_borger", f"{starttime_col}", "date_range"]
```

Comment:
```suggestion
        ["dw_ek_borger", f"{starttime_col}", "date_range"]
```
Move `entity_id` to arguments to make the function more flexible :-)